src/main/java/it/gov/pagopa/pu/migration/wf/activity/ingestion/BaseMigrationFileTypeHandlerActivity.java

```diff
@@ -151,18 +152,25 @@ protected MigrationFileResult handleFilesUpload(
         orElseThrow(() -> new InvalidMigrationFileException("Organization with ipa code " + ipaCodeFile + " not found"));
 
       if (!organization.getBrokerId().equals(upload.getOrganizationId())) {
-        throw new InvalidMigrationFileException("Organization with ipa code " + ipaCodeFile + " is not associated to managed organizations." );
+        throw new InvalidMigrationFileException("Organization with ipa code " + ipaCodeFile + " is not associated to managed organizations.");
       }
       String zipName = Utilities.replaceFileExtension(fileName, ".zip");
       Path zipFilePath = file.getParent().resolve(zipName);
       File zippedFile = zipFileService.zipper(zipFilePath, List.of(file));
       log.info("Processing unzipped file: {}", file);
-      Long id = fileShareService.uploadIngestionFlowFile(
-        organization.getOrganizationId(),
-        ingestionFlowFileType,
-        new FileSystemResource(zippedFile),
-        authnService.getAccessToken(ipaCodeFile)
-      );
+      Long id;
+      try {
+        id = fileShareService.uploadIngestionFlowFile(
+          organization.getOrganizationId(),
+          ingestionFlowFileType,
+          new FileSystemResource(zippedFile),
+          authnService.getAccessToken(ipaCodeFile)
+        );
+      } catch (Exception e) {
```

Comment:
you cannot catch everithing... you should catch just the known not retryable exception!
Actually, the only known use case is the 409 error... you should intercept that exception on the fileshare client and translate it into a NotRetryableException